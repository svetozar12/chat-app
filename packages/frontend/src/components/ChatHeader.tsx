import { Socket } from "socket.io-client";
import React from "react";
import AddGroupChat from "./AddGroupChat";
function ChatHeader({
  sender,
  cookieName,
  socketRef,
}: {
  sender: string;
  cookieName: string;
  socketRef: Socket;
}) {
  return (
    <div className="chat_header" style={{ padding: "1rem" }}>
      <AddGroupChat socketRef={socketRef} cookieName={cookieName} />
    </div>
  );
}

export default ChatHeader;
