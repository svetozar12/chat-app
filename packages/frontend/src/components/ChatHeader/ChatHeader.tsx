import { Socket } from "socket.io-client";
import React from "react";
import AddGroupChat from "components/AddGroupChat/AddGroupChat";
import styled from "@emotion/styled";
function ChatHeader({
  cookieName,
  socketRef,
}: {
  cookieName: string;
  socketRef: Socket;
}) {
  const Header = styled.div`
    box-shadow: 0 0 5px;
    position: relative;
    width: 100%;
    z-index: 9999;
    padding: 1rem;
  `;
  return (
    <Header>
      <AddGroupChat socketRef={socketRef} cookieName={cookieName} />
    </Header>
  );
}

export default ChatHeader;
