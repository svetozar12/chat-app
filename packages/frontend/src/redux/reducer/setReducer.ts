import { Action, ActionType, InitialState2 } from "../state";
export const initialState = {
  reciever: "",
  chat_inviter: "",
  pageNumber: 2,
  setUserSettings: false,
  setFriendRequest: false,
  toggleCreateGroup: false,
  setMobileNav: false,
  setIsMatch: false,
  setChatSettings: false,
  setModalInvite: false,
  setIsLoggingIn: false,
};
const setReducer = (state: InitialState2 | any | InitialState2[] = initialState, action: Action) => {
  switch (action.type) {
    case ActionType.SET_RECIEVER:
      return { ...state, reciever: action.payload };
    case ActionType.CHAT_INVITER:
      return { ...state, chat_inviter: action.payload };
    case ActionType.INCREMENT_PAGE_NUMBER:
      return { ...state, pageNumber: ++action.payload };
    case ActionType.SET_FRIEND_REQUEST:
      return { ...state, setFriendRequest: action.payload };
    case ActionType.SET_USER_SETTINGS:
      return { ...state, setUserSettings: action.payload };
    case ActionType.TOGGLE_CREATE_GROUP:
      return { ...state, toggleCreateGroup: action.payload };
    case ActionType.SET_MOBILE_NAV:
      return { ...state, setMobileNav: action.payload };
    case ActionType.SET_CHAT_SETTINGS:
      return { ...state, setChatSettings: action.payload };
    case ActionType.SET_IS_MATCH:
      return { ...state, setIsMatch: action.payload };
    case ActionType.SET_MODAL_INVITE:
      return { ...state, setModalInvite: action.payload };
    case ActionType.SET_IS_LOGEDIN:
      return { ...state, setIsLoggingIn: action.payload };
    case ActionType.SIGN_OUT:
      return initialState;
    default:
      return state;
  }
};

export default setReducer;
