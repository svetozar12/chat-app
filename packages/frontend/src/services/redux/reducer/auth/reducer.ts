import { HYDRATE } from 'next-redux-wrapper';
import { IAuth } from './state';
import { ActionType } from '../../types';

export const initialState: IAuth = {
  remember_me: false,
  loginPrompt: false,
};
const authReducer = (state = initialState, action: any) => {
  switch (action.type) {
    case ActionType.REMEMBER_ME_CHECK:
      return { ...state, remember_me: action.payload };
    case ActionType.SIGN_IN:
      return { ...state, cookie: action.payload };
    default:
      return state;
  }
};

export default authReducer;
