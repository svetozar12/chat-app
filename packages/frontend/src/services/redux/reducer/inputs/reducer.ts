import IInputs from './state';
import { ActionType } from '../../types';

export const initialState: IInputs = {
  input_username: '',
  input_password: '',
  input_email: '',
  input_gender: 'other',
  input_file: null,
  // notification_number: 0,
};
const inputReducer = (state = initialState, action: any) => {
  switch (action.type) {
    case ActionType.SAVE_INPUT_USERNAME:
      return { ...state, input_username: action.payload };
    case ActionType.SAVE_INPUT_PASSWORD:
      return { ...state, input_password: action.payload };
    case ActionType.SAVE_INPUT_EMAIL:
      return { ...state, input_email: action.payload };
    case ActionType.SAVE_INPUT_GENDER:
      return { ...state, input_gender: action.payload };
    case ActionType.SAVE_INPUT_FILE:
      return { ...state, input_file: action.payload };
    // case ActionType.NOTIFICATION_NUMBER:
    //   return { ...state, notification_number: action.payload };
    default:
      return state;
  }
};

export default inputReducer;
