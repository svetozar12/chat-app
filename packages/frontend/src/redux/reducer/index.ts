import { combineReducers } from "redux";
import reducer from "./authReducer";
import setReducer from "./setReducer";
import saveInputReducer from "./save_inputReducer";
import messageReducer from "./messageReducer";

const reducers = combineReducers({
  authReducer: reducer,
  setReducer,
  messageReducer,
  saveInputReducer,
});

export default reducers;
