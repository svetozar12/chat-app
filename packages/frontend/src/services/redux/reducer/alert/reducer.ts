import { AnyAction } from 'redux';
import { IAlert } from 'services/redux/reducer/alert/state';
import { ActionType } from '../../types';

const initialState: IAlert = {
  good: '', // good and bad stand for good alert and bad alert
  bad: '',
};

const alertReducer = (state = initialState, action: AnyAction) => {
  switch (action.type) {
    case ActionType.LOGIN_POST_ERROR:
      return { ...state, bad: action.payload };
    case ActionType.REGISTER_POST:
      return { ...state, good: action.payload };
    case ActionType.REGISTER_POST_ERROR:
      return { ...state, bad: action.payload };
    default:
      return state;
  }
};

export default alertReducer;
