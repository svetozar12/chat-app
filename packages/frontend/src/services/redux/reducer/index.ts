import { AnyAction, combineReducers } from 'redux';
import authReducer from 'services/redux/reducer/auth/reducer';
import { ActionType } from '../types';
import { HYDRATE } from 'next-redux-wrapper';
import wsReducer from 'services/redux/reducer/websocket/reducer';
import messageReducer from 'services/redux/reducer/messages/reducer';

import { IAuth } from 'services/redux/reducer/auth/state';
import { IWebSocket } from 'services/redux/reducer/websocket/state';
import { IMessage } from 'services/redux/reducer/messages/state';
import IInputs from 'services/redux/reducer/inputs/state';
import inputReducer from 'services/redux/reducer/inputs/reducer';
import toggleReducer from 'services/redux/reducer/toggles/reducer';
import { IToggle } from 'services/redux/reducer/toggles/state';
import IInvite from 'services/redux/reducer/invites/state';
import inviteReducer from 'services/redux/reducer/invites/reducer';
import { IAlert } from 'services/redux/reducer/alert/state';
import alertReducer from 'services/redux/reducer/alert/reducer';

export interface STATE {
  auth: IAuth;
  ws: IWebSocket;
  messages: IMessage;
  inputs: IInputs;
  toggle: IToggle;
  invite: IInvite;
  alert: IAlert;
}

const combiReducers = combineReducers({
  auth: authReducer,
  ws: wsReducer,
  messages: messageReducer,
  inputs: inputReducer,
  toggle: toggleReducer,
  invite: inviteReducer,
  alert: alertReducer,
});

const combReducers = (state: any, action: AnyAction) => {
  if (action.type === HYDRATE)
    return {
      ...state,
      ...action.payload,
    };
  if (action.type === ActionType.SIGN_OUT) {
    return undefined;
  }
  return combiReducers(state, action);
};

export default combReducers;
