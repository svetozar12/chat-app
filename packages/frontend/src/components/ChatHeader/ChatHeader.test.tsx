import ChatHeader from "./ChatHeader";
import renderer from "react-test-renderer";
import { screen, render } from "@testing-library/react";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import { initialState } from "../../redux/reducer/setReducer";

const mockStore = configureStore([]);
const store = mockStore({
  setReducer: initialState,
});
const socketRef: any = jest.fn();

it("should create snapshot for <ChatHeader/>", () => {
  expect(
    renderer
      .create(
        <Provider store={store}>
          <ChatHeader cookieName={"greg"} socketRef={socketRef} />
        </Provider>,
      )
      .toJSON(),
  ).toMatchSnapshot();
});

it("should render <ChatHeader/>", () => {
  render(
    <Provider store={store}>
      <ChatHeader cookieName={"greg"} socketRef={socketRef} />
    </Provider>,
  );
  const renderedComponent = screen.getByTitle("chat_header");
  expect(renderedComponent).toBeInTheDocument();
});
