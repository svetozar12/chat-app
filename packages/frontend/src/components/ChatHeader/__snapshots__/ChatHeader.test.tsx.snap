// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create snapshot for <ChatHeader/> 1`] = `
<div
  className="css-1qx55g5"
  title="chat_header"
>
  <div
    className="css-1h7sx70"
  >
    <form
      className="css-o2j9ze"
      onSubmit={[Function]}
    >
      <input
        className="css-1q3nqrs"
        onChange={[Function]}
        placeholder="Add user ..."
        type="search"
        value=""
      />
      <button
        className="css-csscj8"
        onClick={[Function]}
        type="button"
      >
        Create room
      </button>
    </form>
    <div
      style={
        Object {
          "display": "flex",
        }
      }
    />
  </div>
</div>
`;
